itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -2237,4 +2242,27 @@ private void assertExternalFileInfo(List<String> expected, String dumplocation,
     }
     ReplicationTestUtils.assertExternalFileInfo(warehouseInstance, expected, externalTableInfoFile);
   }
+
+  private void assertFunctionJarsOnTarget(String functionName, List<String> expectedJars) throws IOException {
+    //correct location of jars on target is functionRoot/dbName/funcName/nanoTs/jarFile
+    FileStatus[] fileStatuses = replica.miniDFSCluster.getFileSystem()
+            .globStatus(
+               new Path(replica.functionsRoot + "/" + replicatedDbName.toLowerCase() + "/" +
+                       functionName.toLowerCase() + "/*/*"), path -> path.toString().endsWith("jar")
+            );
+    assertEquals(fileStatuses.length, expectedJars.size());
+    List<String> jars = Arrays.stream(fileStatuses).map(f -> {
+      String[] splits = f.getPath().toString().split("/");
+      return splits[splits.length - 1];
+    }).collect(Collectors.toList());
+    assertThat(jars, containsInAnyOrder(expectedJars.toArray()));
```

Comment:
Does this also do a match on size

Reply:
This is done in a preceding line : assertEquals(fileStatuses.length, expectedJars.size());